From b5af355b1d833b3c898a61cf1e072b59b0ea3440 Mon Sep 17 00:00:00 2001
From: leonsal <leonsal@users.noreply.github.com>
Date: Tue, 3 Dec 2019 16:47:17 -0300
Subject: [PATCH] Fixed issues compiling on Windows with MSVC, CMake and NMake
 (#61)

When trying to build a simple program which uses librtlsdr
as a subproject on Windows, CMake reported several problems
which were solved by:
- Added complete name of libusb in FindLibUSB module.
- Replaced CMAKE_SOURCE_DIR to PROJECT_SOURCE_DIR in src/CMakeLists.txt.
- Replaced header file <afxres.h> in src/rtlsdr.rc.in (only present when windows MFC is
  installed) by <windows.h> which defines the same constants.
---
 cmake/Modules/FindLibUSB.cmake | 2 +-
 src/CMakeLists.txt             | 4 +++-
 src/rtlsdr.rc.in               | 2 +-
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/cmake/Modules/FindLibUSB.cmake b/cmake/Modules/FindLibUSB.cmake
index c2c3f63..bbf5593 100644
--- a/cmake/Modules/FindLibUSB.cmake
+++ b/cmake/Modules/FindLibUSB.cmake
@@ -9,7 +9,7 @@ if(NOT LIBUSB_FOUND)
   )

 #standard library name for libusb-1.0
-set(libusb1_library_names usb-1.0)
+set(libusb1_library_names usb-1.0 libusb-1.0)

 #libusb-1.0 compatible library on freebsd
 if((CMAKE_SYSTEM_NAME STREQUAL "FreeBSD") OR (CMAKE_SYSTEM_NAME STREQUAL "kFreeBSD"))
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8713fba..5d6aff0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,7 +34,7 @@ RTLSDR_APPEND_SRCS(
 # Set up Windows DLL resource files
 ########################################################################
 IF(MSVC)
-    include(${CMAKE_SOURCE_DIR}/cmake/Modules/Version.cmake)
+    include(${PROJECT_SOURCE_DIR}/cmake/Modules/Version.cmake)

     configure_file(
         ${CMAKE_CURRENT_SOURCE_DIR}/rtlsdr.rc.in
diff --git a/src/rtlsdr.rc.in b/src/rtlsdr.rc.in
index bf603c3..0861ecb 100644
--- a/src/rtlsdr.rc.in
+++ b/src/rtlsdr.rc.in
@@ -1,5 +1,5 @@

-#include <afxres.h>
+#include <windows.h>

 VS_VERSION_INFO VERSIONINFO
   FILEVERSION 0,0,0,0
